ORStools/common/client.py

```diff
@@ -87,7 +88,14 @@ def __init__(self, provider=None, agent=None):
 
     overQueryLimit = pyqtSignal()
 
-    def request(self, url, params, first_request_time=None, retry_counter=0, post_json=None):
+    def request(
+        self,
+        url: str,
+        params: dict,
```

Comment:
It depends how far you want to take this, but you can specify the type for the dict keys/values too. Eg `Dict[str, int]`

Reply:
How do I determine how far to take it? I'd guess to base it on readability, which can be judged with evolving experience regarding type hinting?

Reply:
> I'd guess to base it on readability, which can be judged with evolving experience regarding type hinting?

Exactly. There's no science to it, but I tend to document Dict key/value types ONLY if they are simple types (eg int / str).